### PR TITLE
Deprecate qml.load

### PR DESCRIPTION
**Pull request by astralcai**
**Context:**
``qml.load()`` comes up less often in the discussion forum, users seem to find ``from_qiskit`` better in the search

**Description of the Change:**
``qml.load()`` is deprecated and users are directed to the appropriate ``qml.from_x`` function

**Shortcut**
[sc-56056]

### CODE REVIEW

File: pennylane/io.py [hunk 1]
```diff
@@ -77,14 +78,23 @@ def load(quantum_circuit_object, format: str, **load_kwargs):
             Currently, only supported for Qiskit's `QuantumCircuit <https://docs.pennylane.ai/projects/qiskit>`_.
 
     Returns:
-        function: the PennyLane template created from the quantum circuit
-        object
+        function: the PennyLane template created from the quantum circuit object
+
+    .. warning::
+        ``qml.load`` is deprecated and will be removed in version 0.37. Instead, please
+        use the appropriate ``qml.from_x`` functions, such as ``qml.from_qiskit``.
```
Comment (trbromley) [inline pennylane/io.py:85]: Could update this message similarly to the release notes (though this time the reference to the quickstart could actually be a link).

File: pennylane/io.py [hunk 2]
```diff
@@ -77,14 +78,23 @@ def load(quantum_circuit_object, format: str, **load_kwargs):
...
+
     """
 
+    warnings.warn(
+        "qml.load is deprecated and will be removed in version 0.37. Instead, "
+        "please use the appropriate qml.from_x functions, such as qml.from_qiskit.",
```
Comment (trbromley) [inline pennylane/io.py:91]: Since we know the target `format`, we could give a specific recommendation here.

**Review by trbromley — COMMENTED** [review]
Thanks @astralcai!

**Review by lillian542 — APPROVED** [review]
Looks good to me, thanks @astralcai 🚀

File: pennylane/io.py [hunk 3]
```diff
@@ -77,14 +78,24 @@ def load(quantum_circuit_object, format: str, **load_kwargs):
             Currently, only supported for Qiskit's `QuantumCircuit <https://docs.pennylane.ai/projects/qiskit>`_.
 
     Returns:
-        function: the PennyLane template created from the quantum circuit
-        object
+        function: the PennyLane template created from the quantum circuit object
+
+    .. warning::
+        ``qml.load`` is deprecated. Instead, please use the functions outlined in the
+        `Importing workflows <https://docs.pennylane.ai/en/stable/introduction/importing_workflows.html>`_
```
Comment (trbromley) [inline pennylane/io.py:85]: Could this also be a dynamic rather than static link?

File: pennylane/io.py [hunk 4]
```diff
@@ -77,14 +78,24 @@ def load(quantum_circuit_object, format: str, **load_kwargs):
             Currently, only supported for Qiskit's `QuantumCircuit <https://docs.pennylane.ai/projects/qiskit>`_.
 
     Returns:
-        function: the PennyLane template created from the quantum circuit
-        object
+        function: the PennyLane template created from the quantum circuit object
+
+    .. warning::
```
Comment (trbromley) [inline pennylane/io.py:83]: Could we move this warning up to the top, just above the note.

File: pennylane/io.py [hunk 5]
```diff
@@ -77,14 +78,24 @@ def load(quantum_circuit_object, format: str, **load_kwargs):
...
+        quickstart guide, such as ``qml.from_qiskit``.
+
     """
 
+    _format = "pyquil" if format == "pyquil_program" else format
+    warnings.warn(
+        f"qml.load() is deprecated. Instead, please use the more specific qml.from_{_format}()",
+        qml.PennyLaneDeprecationWarning,
+    )
+
```
Comment (trbromley) [inline pennylane/io.py:95]: Nice!

**Review by albi3ro — APPROVED** [review]

**Pull request merged**